floor_generator/lib/misc/string_utils.dart

```diff
@@ -3,4 +3,10 @@ extension StringUtils on String {
   String decapitalize() {
     return '${this[0].toLowerCase()}${substring(1)}';
   }
+
+  /// Flattens a multiline string into a single line string by concatenating
+  /// lines separated by a space and remove leading and trailing whitespace
+  String flatten() {
+    return replaceAll('\n', ' ').replaceAll(RegExp(r'[ ]{2,}'), ' ').trim();
```

Comment:
I'd rather not mess with the query at all and just make the whole CREATE VIEW statement a multiline-string in database_writer.dart

Replacing newlines seems pretty safe, there could be rare cases where a newline is intentionally placed inside a string but I think not permitting that is not an issue, if it is properly documented. But eliminating multiple spaces without knowing the structure of the Query is dangerous. Making the writer output multiline strings seems safer and easier to me (I just tried a bit and it seemed to work)

Reply:
In hindsight, this could also be possible for QueryMethods, but I haven't tried it.

Reply:
Good point. I was wondering if trimming leading and trailing whitespace would be required for SQLite to function as expected and turns out, trimming is not required. This leads to simpler code on our side! 🎉